### ISSUE TITLE
os_xen plugin not working since OPNsense 23.1.2

### ISSUE THREAD

**Issue by billgertz** [opnsense/plugins#3347]
**Important notices**
I acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [x] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [x] The title contains the plugin to which this issue belongs

Please note, this replaces [issue 3440](https://github.com/opnsense/plugins/issues/3340) with a properly formatted issue using a supported template.

**Describe the bug**
Since OPNsense 23.1.2 (amd64 commit a22fc59a5) the os-xen package 1.2_1 no longer correctly runs on boot. There are symlinks missing in the /usr/local/lib directory.

**To Reproduce**
1. From Xen Orchestrator, open the VM General Status tab
2. Verify that the xengust tools are not detected as the line *Management Agent not installed* is displayed
3. Goto Firmware->Packages, and enter "xen" in the search bar
4. Verify that the package shows as "os-xen (installed)"
5. ssh to the OPNsense firewall instance, sudo su, and select 8. Shell to start a cli
6. Attempt to restart the xenguest service:
```
root@opnsense:~ # service xenguest restart
Stopping xenguest.
Waiting for PIDS: 39360.
Starting xenguest.
root@opnsense:~ # ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore-read"
ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore-read"
ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore"
ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore"
ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore"
ld-elf.so.1: Shared object "libxenstore.so.4" not found, required by "xenstore"
.
.
.
```
**To Fix**
1. ssh to the OPNsense firewall instance, sudo su, and select 8. Shell to start a cli
2. Create the missing symlinks and restart the xengust service
```
cd /usr/local/lib
ln -s libxenstore.so.4.0 libxenstore.so.4
ln -s libxenstore.so.4.0 libxenstore.so
ln -s libxentoolcore.so.1.0 libxentoolcore.so.1
ln -s libxentoolcore.so.1.0 libxentoolcore.so

service xenguest restart
```
3. Verify in Xen Orchestra that for the OPNsense firewall, the General Status tab now shows: *Management agent 6.2.0-76888 detected*

**Expected behavior**
Restarting xenguest induces no elf messages about missing libraries.

**Environment**
OPNsense 23.1.2 as VM
XCP-ng 8.2.1
Hardware irrelevant

**Comment by fichtner** [comment]
I'm tagging this "upstream" and closed the other tickets. The FreeBSD ports appears to cause this, but not sure if the port or the original software is at fault. Either way with the little time I have at the moment I can only wait for a fix via FreeBSD ports.

**Comment by fichtner** [comment]
PS: Thanks for posting the workaround. :)

**Comment by billgertz** [comment]
@fichtner Not a problem

**Comment by ClintDavis** [comment]
@billgertz fixed my problem - thanks.

**Comment by theAeon** [comment]
another one on the pile of problem fixed w/ workaround

thx op

**Comment by APz** [comment]
This has now been fixed in upstream.

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=270688

**Comment by fichtner** [comment]
Yes, it will be available with 23.1.6 (likely next week).
